rllib/connectors/connector.py

```diff
@@ -444,17 +449,22 @@ def __getitem__(self, key: Union[str, int, type]):
             elif isinstance(key, int):
                 return [self.connectors[key]]
             elif isinstance(key, type):
-                results = []
+                # Look for subclasses with same name.
```

Comment:
wait, why would the types be different after checkpointing?

Reply:
As per our offline discussion -> https://github.com/ray-project/ray/pull/30635